### PR TITLE
Zy/127 model cluster add configs

### LINKED ISSUE #127: MODEL CLUSTER: Add configs for KMeans
Image

### PR DESCRIPTION
**Pull request by zheyxu**
# Pull Request Details

## What issue does this PR address

- MODEL CLUSTER: Add configs for KMeans

- Link to associated issue: #127 

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Integration test `make qtest.tmp` screenshot included in issue
- [x] Tested on device:
  - [ ] Linux
  - [ ] MacOS
  - [ ] Windows
- [x] Added two reviewers

## Finalising

Once PR discussion is complete and reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

### CODE REVIEW

File: lib/features/cluster/cluster_setting.dart [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+// details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <https://www.gnu.org/licenses/>.
+///
+/// Authors: Zheyuan Xu
+library;
```
Comment (gjwgit) [inline lib/features/cluster/cluster_setting.dart:26]: Should always have a space before `library;`

File: lib/features/cluster/cluster_setting.dart [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        configTopSpace,
+        Row(
+          children: [
+            NumberField(
+              label: 'Clusters:',
+              key: const Key('cluster_number'),
+              tooltip: '''
+
+              The number of clusters (k) or a set of initial (distinct) cluster
+              centers.
```
Comment (gjwgit) [inline lib/features/cluster/cluster_setting.dart:90]: All tooltips should be laid out like this.

File: lib/features/cluster/cluster_setting.dart [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+              ''',
+              controller: _runController,
+              inputFormatter: FilteringTextInputFormatter.digitsOnly,
+              validator: (value) => validateInteger(value, min: 1),
+              stateProvider: clusterRunProvider,
+            ),
+            configWidgetSpace,
+            Checkbox(
+              key: const Key('re_scale'),
```
Comment (gjwgit) [inline lib/features/cluster/cluster_setting.dart]: Missing tooltip.

File: lib/r/source.dart [hunk 4]
```diff
@@ -190,6 +199,17 @@ Future<void> rSource(BuildContext context, WidgetRef ref, String script) async {
   code = code.replaceAll('STEM', stem ? 'TRUE' : 'FALSE');
   code = code.replaceAll('PUNCTUATION', punctuation ? 'TRUE' : 'FALSE');
   code = code.replaceAll('STOPWORD', stopword ? 'TRUE' : 'FALSE');
+  if (!clusterReScale) {
```
Comment (gjwgit) [inline lib/r/source.dart]: I don't want to see R code here - it should be in the R buld script, and choose one approach or the other within the R script.

Comment (gjwgit) [reply lib/r/source.dart]: Why was clusterRescale in r/source.dart placed under the WORDCLOUD transforms? Seems like a very odd place to put it. I moved it to be with CLUSTER below.

**Review by gjwgit — CHANGES_REQUESTED** [review]
I had to:

- [X] Needs a tooltip for Re-Scale
- [X] Please follow the style for tooltip text using triple quotes.
- [X] Style for R named parameters is no space around the ' = '

- [x] See my comment re r/source.dart. The r/model_build_cluster.R script should handle the two options for re-scale, not the dart code.

Why was clusterRescale in r/source.dart placed under the WORDCLOUD transforms? Seems like a very odd place to put it.

**Comment by zheyxu** [comment]
Thanks @gjwgit, the code is updated.

**Review by gjwgit — APPROVED** [review]
Thanks. Looks great.

**Pull request merged**